src/accelerate/commands/launch.py

```diff
@@ -573,7 +573,13 @@ def deepspeed_launcher(args):
         warnings.warn('--fp16 flag is deprecated. Use "--mixed_precision fp16" instead.', DeprecationWarning)
         mixed_precision = "fp16"
 
-    current_env["PYTHONPATH"] = sys.executable
+    def env_path_add(env_var_name, new_path):
+        """extend path-based env variable with a new path"""
+        paths = [p for p in os.environ.get(env_var_name, "").split(":") if len(p) > 0]
+        paths.append(str(new_path))
+        return ":".join(paths)
+
+    current_env["PYTHONPATH"] = env_path_add("PYTHONPATH", sys.executable)
```

Comment:
Let's just write the three lines of code instead of adding a nested function? It will be shorter and as easy to read.

Reply:
as it's likely that at some point you might want to extend other path-based env vars like `PATH` or `LD_LIBRARY_PATH` - one more option is to move it to utils. 

but if you prefer inlined and @pacman100 is an agreement I will rewrite.

Reply:
I'm fine having it as a util too, just don't like the nested function approach where it can't be reused.

Reply:
pushed the change to move to utils, renamed the helper for better clarity now that it's far away from definition.